### PR TITLE
Changed idp enablement to app scheme & ec-301/302 handling

### PR DESCRIPTION
**Pull request by trooper2013**
-  Changed idp enablement to app scheme only.
-  Ec 302 handling
-  Debug logs 
-  Fixed some flapping tests

### CODE REVIEW

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

### Tests: 

Classname | Name|
--- | ---|
SFUserAccountManagerTests | testAuthHandler|
SmartSyncTestSuite | testFetchSObjects|
SmartSyncTestSuite | testSyncSObjectDetectConflictDelete|
SmartSyncTestSuite | testSyncSObjectDetectConflictUpdate|
SmartSyncTestSuite | testSyncSObjectRetrieve|
MetadataManagerTests | testCommonMRUObjectsFromServer|
MetadataManagerTests | testCommonMRUObjectsFromServer|
SyncManagerTests | testCleanResyncGhostsForMRUTarget|
SyncManagerTests | testCleanResyncGhostsForMRUTarget|
SyncManagerTests | testCleanResyncGhostsForMRUTarget|


## Current coverage for SalesforceAnalytics.framework is `67.16%`
✅ *No files affecting coverage found*

---
## Current coverage for SalesforceHybridSDK.framework is `53.66%`
✅ *No files affecting coverage found*

---
## Current coverage for SalesforceSDKCore.framework is `49.59%`
Files changed | - | - 
--- | --- | ---
SFSDKStartURLHandler.m | `0.00%` | :skull:
SFUserAccountManager+URLHandlers.m | `0.00%` | :skull:
SFUserAccountManager+URLHandlers.m | `0.00%` | :skull:
SFSDKStartURLHandler.m | `0.00%` | :skull:
SFSDKIDPAuthClient.m | `1.25%` | :skull:
SFSDKIDPAuthClient.m | `1.25%` | :skull:
SFOAuthCoordinator.m | `35.13%` | :no_entry_sign:
SFOAuthCoordinator.m | `35.13%` | :no_entry_sign:
SFSDKIDPInitCommand.m | `42.86%` | :no_entry_sign:
SFSDKIDPInitCommand.m | `42.86%` | :no_entry_sign:
SalesforceSDKManager.m | `51.93%` | :warning:
SalesforceSDKManager.m | `51.93%` | :warning:
SFUserAccountManager.m | `58.93%` | :warning:
SFUserAccountManager.m | `58.93%` | :warning:
SFSDKOAuthClientConfig.m | `100.00%` | :white_check_mark:
SFSDKOAuthClientConfig.m | `100.00%` | :white_check_mark:

---
## Current coverage for SmartStore.framework is `63.84%`
✅ *No files affecting coverage found*

---
## Current coverage for SmartSync.framework is `79.37%`
✅ *No files affecting coverage found*

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_CI-result">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: native/SampleApps/SmartSyncExplorer/SmartSyncExplorer/Classes/AppDelegate.m [hunk 1]
```diff
@@ -73,9 +73,8 @@ - (id)init
         
         //Uncomment following block to enable IDP Login flow.
         /*
-        [SalesforceSDKManager sharedManager].idpEnabled = YES;
          //scheme of idpAppp
-        [SalesforceSDKManager sharedManager].idpAppScheme = @"sampleidpapp";
+        [SalesforceSDKManager sharedManager].idpAppURIScheme = @"sampleidpapp";
```
Comment (bhariharan) [inline native/SampleApps/SmartSyncExplorer/SmartSyncExplorer/Classes/AppDelegate.m]: Isn't this code commented out in the other sample apps' `AppDelegate` too? If not, we should throw it in there too.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m [hunk 2]
```diff
@@ -211,15 +211,6 @@ - (BOOL)idpEnabled {
     return self.authPreferences.idpEnabled;
 }
 
-- (void)setIdpEnabled:(BOOL)idpEnabled {
-    if (idpEnabled) {
-        [SFSDKAppFeatureMarkers registerAppFeature:kSFSPAppFeatureIDPLogin];
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m]: These feature markers need to be set somewhere - maybe in `setIDPURIScheme`.

File: native/SampleApps/SmartSyncExplorer/SmartSyncExplorer/Classes/AppDelegate.m (hunk 1 above)
Comment (trooper2013) [reply native/SampleApps/SmartSyncExplorer/SmartSyncExplorer/Classes/AppDelegate.m]: Sure can.

File: native/SampleApps/SmartSyncExplorer/SmartSyncExplorer/Classes/AppDelegate.m [hunk 3]
```diff
@@ -157,8 +156,8 @@ - (BOOL)application:(UIApplication *)app openURL:(NSURL *)url options:(NSDiction
     //Uncomment following block to enable IDP Login flow
     /*
     return [[SFUserAccountManager sharedInstance] handleAdvancedAuthenticationResponse:url options:options];
-    */
-   return NO;
+    // */ return NO;
```
Comment (bhariharan) [inline native/SampleApps/SmartSyncExplorer/SmartSyncExplorer/Classes/AppDelegate.m]: The commenting out here is kinda weird. Can we either remove the extra `//` or `/* */`?

File: shared/hybrid/AppDelegate+SalesforceHybridSDK.m [hunk 4]
```diff
@@ -111,15 +125,17 @@ - (void)sfsdk_swizzled_application:(UIApplication *)application didRegisterForRe
...
+    //Uncomment following block to enable IDP Login flow or If you're using advanced authentication:
     // --Configure your app to handle incoming requests to your
     //   OAuth Redirect URI custom URL scheme.
     // --Uncomment the following line and delete the original return statement:
-
-    // return [[SFUserAcountManager sharedInstance] handleAdvancedAuthenticationResponse:url options:options];
-    return NO;
+    /*
+     return [[SFUserAccountManager sharedInstance] handleAdvancedAuthenticationResponse:url options:options];
+     // */ return NO;
```
Comment (bhariharan) [inline shared/hybrid/AppDelegate+SalesforceHybridSDK.m]: The commenting out here is kinda weird. Can we either remove the extra `//` or `/* */`?

File: native/SampleApps/RestAPIExplorer/RestAPIExplorer/Classes/AppDelegate.m [hunk 5]
```diff
@@ -48,6 +48,20 @@ - (id)init
...
+         //user friendly display name
+         [SalesforceSDKManager sharedManager].appDisplayName = @"SampleAppOne";
+         
+         //Use the following code block to replace the login flow selection dialog
+         [SalesforceSDKManager sharedManager].idpLoginFlowSelectionBlock = ^UIViewController<SFSDKLoginFlowSelectionView> * _Nonnull{
+         IDPLoginNavViewController *controller = [[IDPLoginNavViewController alloc] init];
+         return controller;
+         };
+         */
+
```
Comment (bhariharan) [inline native/SampleApps/RestAPIExplorer/RestAPIExplorer/Classes/AppDelegate.m]: Don't you need the commented out code from below here as well?

`return [[SFUserAccountManager sharedInstance] handleAdvancedAuthenticationResponse:url options:options];`

Comment (trooper2013) [reply native/SampleApps/RestAPIExplorer/RestAPIExplorer/Classes/AppDelegate.m]: @bhariharan This code is good. The RestApi Explorer has been set by default to handle advanced Auth. For the IDP case the call is exactly the same.

Comment (bhariharan) [reply native/SampleApps/RestAPIExplorer/RestAPIExplorer/Classes/AppDelegate.m]: Gotcha.

**Pull request merged**